src/main/scala/org/ensime/filewatcher/FileWatchService.scala

```diff
@@ -112,11 +112,11 @@ class FileWatchService {
 
   def registerDir(dir: File, listeners: Set[WatcherListener], wasMissing: Boolean, retry: Int = 2): Unit = {
     if (listeners.exists { l => l.base == dir }) {
-      log.debug(s"delay ${dir} base registration")
+      log.trace(s"delay ${dir} base registration")
```

Comment:
can you please wrap these in `log.isTraceEnabled` ? The overhead can be quite high in tight loops.